src/main/kotlin/frc/robot/autonomous/AutoCommands.kt

```diff
@@ -0,0 +1,95 @@
+package frc.robot.autonomous
+
+import com.pathplanner.lib.auto.AutoBuilder
+import edu.wpi.first.math.controller.PIDController
+import edu.wpi.first.math.geometry.Pose2d
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.math.kinematics.ChassisSpeeds
+import edu.wpi.first.wpilibj.DriverStation
+import edu.wpi.first.wpilibj.DriverStation.Alliance
+import edu.wpi.first.wpilibj2.command.Command
+import edu.wpi.first.wpilibj2.command.Commands
+import frc.robot.CURRENT_MODE
+import frc.robot.Mode
+import frc.robot.lib.distanceFromPoint
+import frc.robot.lib.getPose2d
+import frc.robot.subsystems.drive.Drive
+import frc.robot.subsystems.drive.DriveCommands
+import frc.robot.subsystems.drive.TunerConstants.PATH_CONSTRAINTS
+import frc.robot.swerveDrive
+import frc.robot.vision
+import java.util.function.DoubleSupplier
+import java.util.function.Supplier
+import org.littletonrobotics.junction.Logger
+
+fun alignWithBestVisionTarget(
+    cameraIndex: Int,
+    ySupplier: DoubleSupplier,
+    xSupplier: DoubleSupplier
+): Command =
+    DriveCommands.joystickDriveAtAngle(swerveDrive, ySupplier, xSupplier) {
+        vision.getYawToTarget(cameraIndex)
+    }
+
+fun pathFindToPose(pose: Pose2d): Command =
+    AutoBuilder.pathfindToPoseFlipped(pose, PATH_CONSTRAINTS, 0.0)
+
+fun alignToPose(
+    drive: Drive,
+    robotPose: Supplier<Pose2d>,
+    targetPoseSupplier: Supplier<Pose2d>
+): Command {
+    if (
+        robotPose
+            .get()
+            .distanceFromPoint(targetPoseSupplier.get().translation) >
+            MAX_ALIGNMENT_DISTANCE
+    )
+        return Commands.none()
```

Comment:
`distanceFromPoint()` doesn't do what you think it does. Check its source.
Anyway, you can make it cleaner:
```suggestion
    val robotPose = robotPoseSupplier.get()
    val targetPose = targetPoseSupplier.get()
    if ((robotPose - targetPose).norm > MAX_ALIGNMENT_DISTANCE) {
        return Commands.none()
    }
```
1. Extracted variables
2. Renamed supplier parameter to increase clarity
3. Added braces around if (always good practice)
4. I've also added the following extension property:
    ```kt
    val Transform2d.norm: Distance
        get() = Meters.of(this.translation.norm)
    ```

Reply:
> distanceFromPoint() doesn't do what you think it does. Check its source.

I wrote that extension function, it gives the distance between a pose and a point on the field by converting the pose to a translation and calculating the distance.

Reply:
OK sorry, I had another version of it that had a bug.
Anyway, I think my suggestion is still cleaner

Reply:
I don't think there's a reason to add another extension function when there is already one that exists and works

Reply:
If it's cleaner, uses WPILib terminology (norm) and Kotlin syntax (`>`) and the replacement is very easy (the replacement is written above), I think it's better. You can resolve if you disagree.